services/worker-manager/schemas/v1/worker-lifecycle.yml

```diff
@@ -30,5 +30,21 @@ properties:
       will be terminated.  This value also dictates the lifetime of the
       temporary credentials granted to the worker, meaning that it must be
       less than 30 days. The default value is 4 days.
+  claimTimeout:
+    title: Claim Timeout
+    type: integer
+    minimum: 600 # 10 minutes
+    default: 7200 # 2 hours
+    description: |
+      In order to prevent workers from being stuck without doing any work,
+      `claimTimeout` controls how long a worker can do something other than
+      working on a task. If worker process dies, or it stops calling `claimWork`
+      or `reclaimTask` it should be considered dead and terminated.
+
+      Minimum allowed value is 10 minutes, to give worker chance to start claiming tasks.
```

Comment:
This worries me a _little_ bit. Did you get this 10 min from anywhere or just _hoping_ that this'll be the case? I prefer to go a little more conservative here. Wdyt?

Reply:
yeah, this is totally debatable. I just wanted something to prevent misconfiguration lead to workers being killed before they even got a chance to do something 
I can imagine it takes time to start up and prepare. Also, this would include reboots. On windows it probably takes longer.

@matt-boris do you suggest we increase this number or decrease? 

@petemoore from your past experience, do you know what should be average max time between tasks? GC, reboot, etc

Reply:
> @matt-boris do you suggest we increase this number or decrease?

I would think 30 is a safer bet. This is quite a bit more conservative, which might be overkill, but it's hard to tell for sure.

Reply:
I've been running some [queries](https://sql.telemetry.mozilla.org/queries/92851/source) on the data we have, and noticed that reboots might take 5-15min on average, with some outliers with up to 60+ minutes
Maybe those pools are special, or maybe just the data is not very representative. 

Data source is an aggregation already - total number of seconds spent on reboots and number of instances.. So I'm drawing my conclusions from reboot/instances to see what those numbers are 

Now I'm also wondering if 2h is a good default 🤔

Reply:
ok, if I only query for the pools with 20+ instances, then .75%% gives all reboots within 2h, so I think it should be safe

Reply:
I think there's a difference between a minimum value in the platform and the configured value. If for example we had a provisioner that ran QEMU VMs on demand, 10 minutes might be _far_ too long.  It's probably short for a default value, but not for a _minimum_ value.

It's worth wondering if a host that takes 2+ hours to reboot is going to ever manage to run a build or test task in a reasonable amount of time. But, that decision should be up to the deployers (and thus not encoded in the schema) IMHO.

Reply:
Agree, I'll put something smaller.